near/contracts/bridge/src/lib.rs

```diff
@@ -795,15 +880,65 @@ impl FastBridge {
     /// * The caller does not have any balance.
     #[payable]
     #[pause(except(roles(Role::UnrestrictedWithdraw)))]
-    pub fn withdraw(&mut self, token_id: AccountId, amount: Option<U128>) -> PromiseOrValue<U128> {
-        let recipient_id = env::predecessor_account_id();
-        let user_balance = self.get_user_balance(&recipient_id, &token_id);
+    pub fn withdraw(
+        &mut self,
+        token_id: AccountId,
+        amount: Option<U128>,
+        recipient_id: Option<AccountId>,
+        msg: Option<String>,
+    ) -> Promise {
+        let sender_id = env::predecessor_account_id();
+        self.withdraw_internal(token_id, amount, sender_id, recipient_id, msg)
+    }
+
+    fn withdraw_internal(
+        &mut self,
+        token_id: AccountId,
+        amount: Option<U128>,
+        sender_id: AccountId,
+        recipient_id: Option<AccountId>,
+        msg: Option<String>,
+    ) -> Promise {
+        let user_balance = self.get_user_balance(&sender_id, &token_id);
         let amount = amount.unwrap_or(user_balance);
 
         require!(amount.0 > 0, "The amount should be a positive number");
         require!(amount <= user_balance, "Insufficient user balance");
-        self.decrease_balance(&recipient_id, &token_id, &amount.0);
+        self.decrease_balance(&sender_id, &token_id, &amount.0);
+        let recipient_id = recipient_id.unwrap_or(sender_id);
+
+        if let Some(msg) = msg {
+            self.call_ft_transfer_call(token_id, amount, recipient_id, msg)
+        } else {
+            self.call_ft_transfer(token_id, amount, recipient_id)
+        }
+    }
+
+    fn call_ft_transfer_call(
+        &self,
+        token_id: AccountId,
+        amount: U128,
+        recipient_id: AccountId,
+        msg: String,
+    ) -> Promise {
+        ext_token::ext(token_id.clone())
+            .with_static_gas(utils::tera_gas(50))
+            .with_attached_deposit(1)
+            .ft_transfer_call(recipient_id.clone(), amount, msg)
```

Comment:
should we also add `memo`?

Reply:
The `memo` is useless here. I've just removed it for ft_transfer  here ec6a4f2ed7b5f4ae6d9f38f0b393202c773f4593 because the exact text is generated by the ft itself.